src/closure.rs

```diff
@@ -402,6 +623,56 @@ macro_rules! doit {
                 <&mut Self>::describe();
             }
         }
+
+        #[allow(non_snake_case)]
+        impl<T, $($var,)* R> WasmClosureFnOnce<($($var),*), R> for T
+            where T: 'static + FnOnce($($var),*) -> R,
+                  $($var: FromWasmAbi + 'static,)*
+                  R: ReturnWasmAbi + 'static
+        {
+            type FnMut = FnMut($($var),*) -> R;
+
+            fn into_fn_mut(self) -> Box<Self::FnMut> {
```

Comment:
Technically this isn't quite as optimized as we might otherwise want because a `FnOnce` that doesn't actually close over anything will allocate because of the discriminant on `Option<T>`. I think fully supporting this in the limit though will require specialized support in the JS bindings (the JS bindings are the ones to clear the stored pointer).

Mind throwing a comment somewhere about that? I think this is totally fine as a first pass

Reply:
(technically the same comment for the one below too)